File: ga/4omini_result_2024_7/result/1968C_mini/43_1968C.c
use std::io;

const N: usize = 500;
const A: usize = 1_000_000_000;

fn proc(n: usize, x: &[usize]) {
    let mut a = vec![0; N];

    a[0] = x[1] + 1;
    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j * a[i - 1];
            if (i + 1 < n) as usize + 1 < n && (x[i + 1] < a[i]) && (a[i] != a[i - 1]) {
                break;
            }
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let n: usize = input.trim().parse().unwrap();
        let mut x = vec![0; N];

        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let nums: Vec<usize> = input.split_whitespace()
            .map(|s| s.parse().unwrap())
            .collect();

        for (i, &num) in nums.iter().enumerate() {
            x[i + 1] = num; // Fill x starting from index 1
        }

        proc(n, &x);
    }
}

File: ga/glm4_result_2024_6/result/1968C_glm/43_1968C.rs
use std::io::{self, BufRead};

fn proc() {
    let stdin = io::stdin();
    let mut line = String::new();
    stdin.read_line(&mut line).unwrap();
    let n: usize = line.trim().parse().unwrap();
    let mut x = vec![0; n];
    for i in 1..n {
        let mut line = String::new();
        stdin.read_line(&mut line).unwrap();
        x[i] = line.trim().parse().unwrap();
    }
    let mut a = vec![0; n];
    a[0] = x[1] + 1;
    for i in 1..n {
        let mut j = 0;
        loop {
            a[i] = x[i] + j * a[i - 1];
            if i + 1 < n && x[i + 1] < a[i] && a[i] != a[i - 1] {
                break;
            }
            j += 1;
        }
    }
    for i in a {
        print!("{} ", i);
    }
    println!();
}

fn main() {
    let stdin = io::stdin();
    let mut line = String::new();
    stdin.read_line(&mut line).unwrap();
    let t: usize = line.trim().parse().unwrap();
    for _ in 0..t {
        proc();
    }
}